### ISSUE TITLE
Configure queue per task

### ISSUE THREAD

**Issue by adamzapasnik** [Shopify/maintenance_tasks#777]
I don't think it's possible right now, is it? My reasoning for such option is:
1. some tasks run data migrations that can take hours/days and we're fine with them running in the lowest priority queue.
2. but some tasks are "crucial" and they run within a few minutes and we expect them to run immediately after being ran/enqueued via UI, thus we need to run them in the highest priority queue. An example: some data from a 3rd party API is out of sync, we'd like to sync it immediately via a maintenance task.

Does it make sense? Or maintenance_tasks wasn't really meant for such cases? 
```[tasklist]

```

**Comment by nvasilevski** [comment]
>Does it make sense?

It does! But one thing I would do is to look on the issue at a higher level and consider allowing `MaintenanceTasks.job` to be configured on per-task basis. The reason for this is that `queue` is a concept from background processing and not from maintenance tasks itself so it would be better for the gem not to know about queues as nothing really enforces the job that executes tasks to even have a queue.
Despite of the gem heavily mentioning `Job` concept and providing `TaskJob` as basically the only native way to execute a task, I still see it as an "executor" abstraction. By that I mean that technically we still can configure `MaintenanceTasks.job = ` to be `"MyCustomClassThatIsTotallyNotAJob"` and the gem will work as long as `MyCustomClassThatIsTotallyNotAJob` string can be constantized, respond to `new`, instances of the class respond to `enqueue(run)` and the class knows how to handle `run`. Using a non-job based executor sounds unnecessary but still possible design-wise as long as the executor is abstracted well enough.

To summarize, to me it makes total sense if we look on it from the per-task executor point of view. If a single task can specify custom executor (a job, for most cases) then this task can be executed within a custom context, whether it's a separate queue, or specific parallel execution constraints, or custom retries policy, the `maintenance_task` gem itself won't need to know about these background-execution concepts
